packages/Adapters/Intrepid/src/DTK_IntrepidEntityIntegrationRule.cpp

```diff
@@ -0,0 +1,99 @@
+//---------------------------------------------------------------------------//
+/*
+  Copyright (c) 2012, Stuart R. Slattery
+  All rights reserved.
+
+  Redistribution and use in source and binary forms, with or without
+  modification, are permitted provided that the following conditions are
+  met:
+
+  *: Redistributions of source code must retain the above copyright
+  notice, this list of conditions and the following disclaimer.
+
+  *: Redistributions in binary form must reproduce the above copyright
+  notice, this list of conditions and the following disclaimer in the
+  documentation and/or other materials provided with the distribution.
+
+  *: Neither the name of the University of Wisconsin - Madison nor the
+  names of its contributors may be used to endorse or promote products
+  derived from this software without specific prior written permission.
+
+  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+  A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+  HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+//---------------------------------------------------------------------------//
+/*!
+ * \brief DTK_IntrepidEntityIntegrationRule.cpp
+ * \author Stuart R. Slattery
+ * \brief Intrepid integration rule implementation.
+ */
+//---------------------------------------------------------------------------//
+
+#include "DTK_IntrepidEntityIntegrationRule.hpp"
+#include "DTK_IntrepidHelpers.hpp"
+
+#include <Shards_CellTopology.hpp>
+
+#include <Intrepid_FieldContainer.hpp>
+
+namespace DataTransferKit
+{
+//---------------------------------------------------------------------------//
+// Given an entity and an integration order, get its integration rule. 
+void IntrepidEntityIntegrationRule::getIntegrationRule(
+    const shards::CellTopology& entity_topo,
+    const int order,
+    Teuchos::Array<Teuchos::Array<double> >& reference_points,
+    Teuchos::Array<double>& weights ) const
+{
+    // If we haven't already created a cubature for this topology and order
+    // create one.
+    std::pair<unsigned,int> cub_key( entity_topo.getKey(), order );    
+    Teuchos::RCP<Intrepid::Cubature<double> > cub_rule;
+    if ( d_cub_rules.count(cub_key) )
+    {
+	cub_rule = d_cub_rules.find( cub_key )->second;
+    }
+    else
+    {
+	cub_rule = d_intrepid_factory.create( entity_topo, order );
```

Comment:
Why it is that `DefaultCubatureFactory::create(...)` is not declared `const`? I wish we hadn't to declare the factory `mutable`.